### PR TITLE
Adds the M1911 + Adds .45 ACP as a new calibre.

### PR DESCRIPTION
**Pull request by BrotherHangyul**
About The Pull Request:

Adds the "M1911" handgun, chambered in .45 ACP and holding seven rounds. Currently the only gun capable of using .45 ACP. Probably coded horribly but I tried my best okay?

.45 ACP is a bit stronger than .40 Automag but weaker than .50. Due to the power of the round the gun cannot be silenced, and the M1911 has no sort of extended magazine, leaving it at seven rounds and nothing else. I advise that any guns in the future using .45 ACP either have incredibly low accuracy, magazine size or fire-rate to make up for the genuine power of the round.

The M1911 is basically the "Colt" but chambered in .45 ACP. Holds seven rounds. The sprite is basically just the colt sprite but the grip is a bit darker.

This is my first time coding a proper, not half-assed PR, please don't crucify me.

This has actually been tested, and it works, at least on my device. Actual implementation of the M1911 and the .45 ACP calibre will be done in the future once I regain the energy, it'll likely be available to Blackshields first before anyone else.

Screenshot (127)

## Changelog
:cl:
add: M1911 handgun.
add: .45 ACP calibre and rounds.
add: .45 magazines.
add: coltreal sprite, basically just the colt sprite but a bit darker.
/:cl:

### CODE REVIEW

**Comment by BrotherHangyul** [comment]
Forgot to add:

Being an aged gun, the M1911 does not have auto-eject. This is all for balance, as .45 ACP actually does clap pretty hard.

**Comment by r4d6** [comment]
While a new gun is always good, I don't recommend adding yet another caliber, we have enough of those around.

Instead what you can do instead is to have the gun use either .40 or .50 Kurtz, and adjust the gun's damage mods to compensate.

**Comment by BrotherHangyul** [comment]
Bug:

- M1911 can use .35 Auto
- Most handguns can equip .45 ACP, though the 7-round capacity stays the same.

I am currently fixing the issue.

**Comment by BrotherHangyul** [comment]
- For the moment, .40 and .45 will have to share CAL_MAGNUM. This means that guns using .40 can equip .45 magazines, **however**, as the M1911 and .45 magazines are currently unobtainable outside of adminspawn, this shouldn't be an issue that effects the game (if merged) for a while. I am currently looking for a permanent fix.

**Comment by BrotherHangyul** [comment]
Update:

- This is finally fixed, the define for .45 ACP is now CAL_45.

**Comment by BrotherHangyul** [comment]
I will be adding more weapons using .45 ACP in the future, this is more or less the framework for these weapons. As is stated by the coding guidelines, multiple-part projects are preferable to a large one that flops. I will be adding a larger variety of sidearms (revolvers and handguns) and burst-fire weapons using .45 ACP when I am more capable of doing so.

**Comment by ThatGuyFromThatPlace** [comment]
Isn't this the 1911 sprite already being used for the basic .35 Colt pistol?

Also, there used to be a lot more ammo types on Sojourn, but, they were intentionally removed and/or two types merged into single rounds. 
https://github.com/sojourn-13/sojourn-station/pull/94
Consider this beforehand.

"FS HG .45 Bronco" was changed to BR .35 \"Bronco\"
H&S HG 10mm \"Delta Elite\" was changed to H&S HG .40 \"Delta Elite\"
Etc. etc.

**Comment by BrotherHangyul** [comment]
> Isn't this the 1911 sprite already being used for the basic .35 Colt pistol?
Yes, I have modified it slightly to still look like the 1911 whilst being different enough to tell the two apart (The .45 ACP M1911 has a darker grip and gunmetal compared to the .35 variant).
> Also, there used to be a lot more ammo types on Sojourn, but, they were intentionally removed and/or two types merged into single rounds. #94 Consider this beforehand.
> 
> "FS HG .45 Bronco" was changed to BR .35 "Bronco" H&S HG 10mm "Delta Elite" was changed to H&S HG .40 "Delta Elite" Etc. etc.

This PR was merged a year and a half ago, I think a little variation will do some good in the world.

File: code/modules/projectiles/ammunition/bullets.dm [hunk 1]
```diff
@@ -114,6 +114,57 @@
 /obj/item/ammo_casing/magnum/scrap/prespawned
 	amount = 6
 
+//// .45 ACP ////
+
+/obj/item/ammo_casing/pistol_45
+	desc = "A .45 ACP bullet casing."
+	icon_state = "pistol_c_l"
+	spent_icon = "pistol_c_l-spent"
+	caliber = CAL_PISTOL
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/ammunition/bullets.dm:123]: Swap all these to CAL_45 to fix the pistole bug

Comment (BrotherHangyul) [reply code/modules/projectiles/ammunition/bullets.dm]: Should be fixed.

**Comment by Kazkin** [comment]
While not a bad idea, we are very very **VERY** reluctant to add a new caliber type for what amounts to a single gun that uses it. I would say, pretty definitively, that you'll want to make it a .40 caliber, given we specifically removed the .45 some time ago.

**Comment by BrotherHangyul** [comment]
> While not a bad idea, we are very very **VERY** reluctant to add a new caliber type for what amounts to a single gun that uses it. I would say, pretty definitively, that you'll want to make it a .40 caliber, given we specifically removed the .45 some time ago.

As previously stated, I will be adding more guns using .45 in the future should the PR be merged, and this also serves as framework for anyone else wishing to do so. The variety in handgun calibres is rather weak at the moment; .35 is basic, .40 is somewhat unique and .50 Kurtz is both rather inaccessible and just a simple "uber" round. I also believe that following the increments of five would be somewhat beneficial (.35, .40, .45, .50). Adding a bus of guns onto this PR at the start would cause errors, complications and different issues for each individual gun, whether it be balance or code-wise. This could be considered a part 1 of a project.

**Comment by Kazkin** [comment]
I get your reasoning, I'm just extremely hesitant to add ammo, as the last time we removed/added an ammo type it caused alot of issues both from the standpoint of gameplay and in one case code wise.

What I'm considering is why removed .45 ammo in the first place, that is, that adding ammo complexity through different calibers has long reaching effects on the core mechanic of the server, the loot and shoot gameplay that people do outside of roleplay. Consider that, we add a full slew of .45 ammo from pistols to rifles, this creates a scenario where a colonist or outside using a weapon who then finds said ammo must match it. A prospector using a 20mm shotgun and .40 magnum pistol needs these two ammo types. By adding .45 ammo we are reducing the chance of that player finding the relevant ammo they use.

Consider that even if the ammo has equal drop rates between any pistol ammo spawns (.35, .40, .50) your reducing the odds of a person finding their relevant ammo from 1/3rd to 1/4th. This in turn makes ammo harder to stretch across supply lines for scavengers, making them more reliant on alternative choices they may not be built for. Now consider ammo spawns may also be greater variants (.35, .40, .50, 20mm, .257, 7.5). If you add .45 the odds of now getting **any** ammo type you need goes from 1/6th to 1/7th even if that person is carrying a .45 weapon.

Now consider existing weapons. We have models of the colt m1911 that comes in the .35 and .40 variants in 2 separate versions. We have the Colt or Bronco in .35 and Delta Elite or Stallion in .40 caliber. You're sprite is good, I like it, and I'm glad your showing interest in code, in particular code that is an expanding scope. However, in this circumstance, I cannot say that adding a new caliber for a 5th variant of a pistol is the best plan from a mechanics standpoint. In particular for the far reaching gameplay changes this would make, not just in the terms of looting but supply lines and variants such as laser ammo, ammo printer by the munitions fabricator, ammo obtained through cargo, uplink ammo for antags, and ammo supplied at round start for combative roles.

**Comment by Kazkin** [comment]
To add on an additional bit of thoughts after considering it further, I think of the special/snowflake calibers we have added as well. The two cases that come to mind are the 70mm rolled shot and crossbow bolts for the baroque hunting rifle and bolt lancer respectively. In the circumstances of these weapons and their caliber, the importance to me comes down to accessibility. In both cases these ammo types are to 1 weapon that is only used by 1 job and cannot be found/scavenged anywhere. **However**, the ammo is extremely accessible regardless, as hunters can craft crossbow bolts using only metal rods, while ammo kits can be crafted by anyone, allowing for a steady and ready supply of 70mm rolled shot. This is important because it means that, when this was added, this ammo calibers did not disrupt the loot pools for munitions nor did it affect other jobs. They were unique, directly effecting specific roles, and covering a limited scope with systems in place to ensure limited, if any, scarcity. Unique caliber types should, reasonably, follow this model. While widely use calibers, even if they have multiple guns, need to be limited in their berth to prevent players from having widely limiting weaponry.

A blackshield or prospector who has plenty of .40 ammo for their sidearm will be reluctant to swap said sidearm to a different caliber, as it makes all the ammo they have useless. Players will naturally shop for upgraded weaponry, but often if they have ammo for a certain caliber, go for that caliber type. If I have a crappy rose pistol (.40 caliber) and 40 bullets, I'd want to upgrade to a better .40 magnum pistol, as I wouldn't wish to or possibly might not have the capacity to sell/swap to a new ammo type like .35/.50 or in this case .45 ammo. My limited funds can upgrade me to a pilgrim which I have ammo for, but not a colt with 40 bullets of .45 ammo.

**Pull request closed without merging**